Realm/ObjectServerTests/RLMSyncTestCase.m

```diff
@@ -97,14 +97,16 @@ - (void)waitForDownloadsForUser:(RLMSyncUser *)user
     NSAssert(realms.count == counts.count && realms.count == realmURLs.count,
              @"Test logic error: all array arguments must be the same size.");
     XCTestExpectation *checkCountExpectation = [self expectationWithDescription:@"Downloads should complete"];
-    for (NSUInteger i = 0; i < realms.count; i++) {
-        WAIT_FOR_DOWNLOAD(user, realmURLs[i]);
-    }
     dispatch_async(dispatch_get_main_queue(), ^{
         for (NSUInteger i = 0; i < realms.count; i++) {
-            CHECK_COUNT([counts[i] integerValue], SyncObject, realms[i]);
+            WAIT_FOR_DOWNLOAD(user, realmURLs[i]);
         }
-        [checkCountExpectation fulfill];
+        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
How does this `dispatch_async` help? Seems like refreshing the realms that need an update would be a more appropriate way to do this?